core/src/subgraph/context/instance.rs

```diff
@@ -155,7 +156,36 @@ where
         })
     }
 
-    pub(super) fn revert_data_sources(&mut self, reverted_block: BlockNumber) {
+    /// Reverts any DataSources that have been added from the block forwards (inclusively)
+    /// This function also reverts the done_at status if it was 'done' on this block or later.
+    /// It only returns the offchain::Source because we don't currently need to know which
+    /// DataSources were removed, the source is used so that the offchain DDS can be found again.
+    pub(super) fn revert_data_sources(
+        &mut self,
+        reverted_block: BlockNumber,
+    ) -> Vec<offchain::Source> {
+        self.revert_quick(reverted_block);
+
+        // The following code handles resetting offchain datasources so in most
+        // cases this is enough processing.
+        // At some point we prolly need to improve the linear search but for now this
+        // should be fine. *IT'S FINE*
```

Comment:
The `*IT'S FINE*` is very reassuring but perhaps a bit too reassuring 😄

Reply:
![image](https://user-images.githubusercontent.com/1640325/200579553-1199a7db-3c90-4b23-9b70-f66a21f49e77.png)